src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,133 @@
+package duke;
+
+import duke.error.IllegalInputException;
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.Todo;
+
+import java.util.Scanner;
+import java.util.ArrayList;
+import java.io.FileWriter;
+import java.io.IOException;
+
+public class Duke {
+    public static void main(String[] args) {
+
+        String input;
+        ArrayList<Task> taskList = new ArrayList<>();
+
+        printWelcomeMessage();
+
+        // Echo loops until "bye" is inputted
+        while(true) {
+            Scanner echo = new Scanner(System.in); // Scan user input
+            input = echo.nextLine();
+
+            // Exit program if "bye" is inputted
+            try {
+                if (input.length() == 0) {
+                    throw new IllegalInputException();
+                } if (input.equals("bye")) {
+                    System.out.println("Goodbye. See you soon!");
+                    break;
+                } else if (input.equals("list")) {
+                    printList(taskList);
+                } else if (input.contains("done")) {
+                    markAsDone(input, taskList);
+                } else if (input.contains("delete")){
+                    deleteTask(input, taskList);
+                } else{
+                    addTask(input, taskList); // Add input as new task
+                }
+                saveTasks("saveFile.txt", taskList);
+            } catch (IllegalInputException e) {
+                System.out.println("Input cannot be empty!");
+            }
+        }
+    }
+
+    private static void addTask(String input, ArrayList<Task> taskList) {
+        try {
+            if (input.contains("deadline")) {
+                int slashLocation = input.indexOf("/");
+                taskList.add(new Deadline(input.substring(input.indexOf("n") + 3, slashLocation), input.substring(slashLocation + 1)));
```

Comment:
Perhaps you can try separating the variables within this Deadline object to improve code readability.